ui/eureka/eureka.js

```diff
@@ -2074,28 +2550,33 @@ class EurekaTracker {
   OnLog(e) {
     if (!this.zoneInfo)
       return;
-    for (let idx = 0; idx < e.detail.logs.length; idx++) {
-      let log = e.detail.logs[idx];
-      let gRegex = this.TransByParserLang(this.options.Regex);
-      let gFlagRegex = gRegex['gFlagRegex'];
+    for (const log of e.detail.logs) {
+      let gFlagRegex = this.TransByParserLang(this.options.Regex, 'gFlagRegex');
       let match = log.match(gFlagRegex);
       if (match)
         this.AddFlag(match[2], match[3], match[1], match[4]);
-      let gTrackerRegex = gRegex['gTrackerRegex'];
+
+      if (this.fairy) {
+        if (log.includes(' 03:') || log.includes('00:0839:')) {
+          match = log.match(this.fairy.regex);
+          if (match)
+            this.AddFairy(match.groups);
+        }
+      }
```

Comment:
this one can be flattened
```suggestion
      if (this.fairy && (log.includes(" 03:") || log.includes("00:0839:"))) {
        match = log.match(this.fairy.regex);
        if (match) this.AddFairy(match.groups);
      }
```

Reply:
I think the previous code is more readable here, even if it's longer.